puppet/chassis.rb

```diff
@@ -177,7 +177,24 @@ def self.install_extension(extension)
 			repo = extension
 		end
 
-		folder = @@dir + '/extensions/' + extension.split('/').last.gsub(/.git$/, '').downcase
+		# We need to add a version to allow for a seamless upgrade to Bionic Beaver.
+		if config["version"] >= 3
+			# Check for existing extensions that have a hyphen and remove them.
+			old_extensions = Dir.glob(@@dir + '/extensions/*')
+			old_extensions.each { |extension|
+				if extension.include? "-"
```

Comment:
Could we `continue unless extension.include? "-"`?

Reply:
Ahh good thinking. In Ruby `continue` is `next` so I've added that.